apps/src/music/utils/Chords.ts

```diff
@@ -0,0 +1,120 @@
+import {ChordEventValue} from '../player/interfaces/ChordEvent';
+
+// This file contains helper functions for chords, and is used by the
+// block's custom field, the custom field editor panel, and the music player.
+
+// A single note to be played in a chord.
+export interface ChordNote {
+  tick: number;
+  note: number;
+}
+
+// A single note from a chord to be rendered in a graph.
+export interface ChordGraphNote {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+// Generate an array containing tick numbers from 1..16.
+const arrayOfTicks = Array.from({length: 16}, (_, i) => i + 1);
+
+// Given a ChordEventValue, generate a set of notes.
+export function generateNotesFromChord(
+  chordEventValue: ChordEventValue
+): ChordNote[] {
+  const {notes, playStyle} = chordEventValue;
+  if (notes.length === 0) {
+    return [];
+  }
+
+  if (playStyle === 'arpeggio-up') {
+    notes.sort();
+  } else if (playStyle === 'arpeggio-down') {
+    notes.sort().reverse();
+  } else if (playStyle === 'arpeggio-random') {
+    // Randomize using Fisher-Yates Algorithm
+    for (let i = notes.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      const temp = notes[i];
+      notes[i] = notes[j];
+      notes[j] = temp;
+    }
+  }
+
+  if (playStyle === 'arpeggio-random') {
+    return arrayOfTicks.map(tick => {
+      return {
+        tick: tick,
+        note: notes[Math.floor(Math.random() * notes.length)]
+      };
+    });
+  } else if (playStyle === 'together') {
+    return notes.map(note => {
+      return {
+        tick: 1,
+        note
+      } as ChordNote;
+    });
+  } else {
+    if (playStyle === 'arpeggio-up') {
+      notes.sort();
+    } else if (playStyle === 'arpeggio-down') {
+      notes.sort().reverse();
+    }
+
+    return arrayOfTicks.map(tick => {
+      return {
+        tick: tick,
+        note: notes[(tick - 1) % notes.length]
+      };
+    });
+  }
+}
+
+// Given a ChordEventValue, generate a set of data for graphing it.
+export function generateGraphDataFromChord(options: {
+  chordEventValue: ChordEventValue;
+  width: number;
+  height: number;
+  numOctaves: number;
+  startOctave: number;
+  padding: number;
+  noteHeightScale: number;
+}): ChordGraphNote[] {
+  const {
```

Comment:
minor: I think if you pull out the structure of `options` into and interface, you can split up the arguments in the function signature directly (`generateGraphDataFromChord({chordEventValue, width...}: Options)`) - though that would still end up with the same lines of code, so either way is fine 🙂

Reply:
Oh that does sound more modern though.  Will do!